python_modules/dagster/dagster/_core/definitions/repository_definition.py

```diff
@@ -1379,6 +1416,78 @@ def __call__(self, *args, **kwargs):
         return self
 
 
+class PendingRepositoryDefinition:
+    def __init__(
+        self,
+        name: str,
+        repository_definitions: Sequence[
+            Union[RepositoryListDefinition, "CacheableAssetsDefinition"]
+        ],
+        description: Optional[str] = None,
+        default_logger_defs: Optional[Mapping[str, LoggerDefinition]] = None,
+        default_executor_def: Optional[ExecutorDefinition] = None,
+    ):
+        self._repository_definitions = check.list_param(
+            repository_definitions,
+            "repository_definition",
+            additional_message="PendingRepositoryDefinition supports only list-based repository data at this time.",
+        )
+        self._name = name
+        self._description = description
+        self._default_logger_defs = default_logger_defs
+        self._default_executor_def = default_executor_def
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    def get_repository_load_data(self) -> RepositoryLoadData:
```

Comment:
`build` / `fetch` / `compute` - to telegraph expected weight of execution, `get` implies cheap/free (at least to me)

Reply:
nice yeah